snap/local/start-mysqld.sh

```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
 exec $SNAP/usr/bin/setpriv --clear-groups --reuid snap_daemon \
-  --regid root -- $SNAP/usr/bin/mysqld_safe --defaults-file=$SNAP_DATA/etc/mysql/mysql.cnf "$@"
+  --regid snap_daemon -- $SNAP/usr/bin/mysqld_safe --defaults-file=$SNAP_DATA/etc/mysql/mysql.cnf "$@"
```

Comment:
i would be careful with this change -> especially when creating/restoring backups ([this may be relevant](https://github.com/canonical/mysql-operator/blob/main/src/mysql_vm_helpers.py#L374-L375))